spring-integration-jdbc/src/main/java/org/springframework/integration/jdbc/lock/JdbcLockRegistry.java

```diff
@@ -113,7 +115,8 @@ public void lock() {
 					}
 					break;
 				}
-				catch (CannotSerializeTransactionException | TransactionTimedOutException | QueryTimeoutException e) {
+				catch (CannotSerializeTransactionException | TransactionTimedOutException | QueryTimeoutException |
```

Comment:
The `CannotSerializeTransactionException` and `DeadlockLoserDataAccessException` are both inheritors of the `PessimisticLockingFailureException`, so maybe we can go the way for their super class?

Reply:
After looking at the exception hierarchy - maybe  we can use `TransientDataAccessException`? It does make sense: any transient error should cause retry.